iris-mpc-cpu/src/hawkers/galois_store.rs

```diff
@@ -151,22 +181,83 @@ impl VectorStore for LocalNetAby3NgStoreProtocol {
+            let vector_point = storage.points[vector.val()].clone();
+            let pairs = vec![(query_point.data, vector_point.data)];
+            jobs.spawn(async move {
+                let ds_and_ts = eval_pairwise_distances(pairs, &mut player_session).await;
+                DistanceShare {
+                    code_dot: ds_and_ts[0].clone(),
+                    mask_dot: ds_and_ts[1].clone(),
+                    player:   player.clone(),
+                }
+            });
+        }
+        jobs.join_all().await
     }
 
-    async fn is_match(&self, distance: &Self::DistanceRef) -> bool {
+    async fn eval_distance_batch(
+        &self,
+        query: &Self::QueryRef,
+        vectors: &[Self::VectorRef],
+    ) -> Vec<Self::DistanceRef> {
         let ready_sessions = self.runtime.create_player_sessions().await.unwrap();
         let mut jobs = JoinSet::new();
         for player in self.runtime.identities.clone() {
             let mut player_session = ready_sessions.get(&player).unwrap().clone();
             let storage = self.players.get(&player).unwrap();
-            let x = storage.points[distance.0.val()].clone();
-            let mut y = storage.points[distance.1.val()].clone();
-            y.data.code.preprocess_iris_code_query_share();
-            y.data.mask.preprocess_mask_code_query_share();
+            let query_point = storage.points[query.val()].clone();
+            let pairs = vectors
+                .iter()
+                .map(|vector_id| {
+                    let vector_point = storage.points[vector_id.val()].clone();
+                    (query_point.data.clone(), vector_point.data)
+                })
+                .collect::<Vec<_>>();
             jobs.spawn(async move {
-                galois_ring_is_match(&mut player_session, &[(x.data, y.data)])
+                let ds_and_ts = eval_pairwise_distances(pairs, &mut player_session).await;
+                ds_and_ts
+                    .chunks(2)
+                    .map(|dot_products| DistanceShare {
+                        code_dot: dot_products[0].clone(),
+                        mask_dot: dot_products[1].clone(),
+                        player:   player.clone(),
+                    })
+                    .collect::<Vec<_>>()
+            });
+        }
+        // Now we have a vector of 3 vectors of DistanceShares, we need to transpose it
+        // to a vector of DistanceRef
+        let mut all_shares = jobs
+            .join_all()
+            .await
+            .into_iter()
+            .map(|player_shares| player_shares.into_iter())
+            .collect::<Vec<_>>();
+        (0..vectors.len())
+            .map(|_| {
+                all_shares
+                    .iter_mut()
+                    .map(|player_shares| player_shares.next().unwrap())
+                    .collect::<Self::DistanceRef>()
+            })
+            .collect::<Vec<Self::DistanceRef>>()
```

Comment:
Just to be clear, this is just an artifact of the local variant where all players are done by this party and this would not happen in a real network right?